scipy/stats/tests/test_survival.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_equal
+from scipy import stats
+
+class TestSurvival:
+    def test_input_validation(self):
+        message = '`sample` must be a one-dimensional sequence.'
+        with pytest.raises(ValueError, match=message):
+            stats.ecdf([[1]])
+        with pytest.raises(ValueError, match=message):
+            stats.ecdf(1)
+
+        message = '`sample` must not contain nan'
+        with pytest.raises(ValueError, match=message):
+            stats.ecdf([np.nan])
+
+        message = 'Currently, only uncensored data is supported.'
+        with pytest.raises(NotImplementedError, match=message):
+            stats.ecdf(stats.CensoredData.left_censored([1], censored=[True]))
+
+    def test_edge_cases(self):
+        res = stats.ecdf([])
+        assert_equal(res.x, [])
+        assert_equal(res.cdf, [])
+
+        res = stats.ecdf([1])
+        assert_equal(res.x, [1])
+        assert_equal(res.cdf, [1])
+
+    def test_unique(self):
+        # Example with unique observations; `stats.ecdf` ref. [1] page 80
+        sample = [6.23, 5.58, 7.06, 6.42, 5.20]
+        res = stats.ecdf(sample)
+        ref_x = np.sort(np.unique(sample))
```

Comment:
One thing I was not sure here. Because we use `unique` we won't have the same number of output values as the input. I am fine with that since we return both `x` and `cdf`

Reply:
I don't understand. `ecdf` also takes the unique values.  It returns only the unique values in `sample`, and it returns a CDF associated with each.

Reply:
You could have had this
```
xs = np.sort(data)
ys = np.linspace(0, 1, num=len(data))
```